xcube-dev/xcube
# New server `statistics` endpoint

Added new statistics API to xcube server. The service computes basic statistical values and a histogram for given data variable, time stamp, and a GeoJSON geometry. Its endpoint is: `/statistics/{datasetId}/{varName}?time={time}`. Geometry is passed as request body in form of a GeoJSON geometry object.

To verify:

1. In xcube project folder, run xcube server with the demo configuration:  `xcube --loglevel=INFO --traceback serve --traceperf -vvv -c examples/serve/demo/config.yml`
2. Open `http://127.0.0.1:8080/openapi.html` to verify the API is there
3. Open `http://127.0.0.1:8080/viewer` and try the new `Statistics`tab in the right sidebar.

The current state of https://github.com/xcube-dev/xcube-viewer/pull/357 has been bundled in this PR.

Checklist:

* [x] Add unit tests and/or doctests in docstrings
* [x] Add docstrings and API docs for any new/modified user-facing classes and functions
* [x] New/modified features documented in `docs/source/*`
* [x] Changes documented in `CHANGES.md`
* [x] GitHub CI passes
* [x] AppVeyor CI passes
* [x] Test coverage remains or increases (target 100%)
